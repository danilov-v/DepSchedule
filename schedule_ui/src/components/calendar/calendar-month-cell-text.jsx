import React from "react";
import PropTypes from "prop-types";
import { CELL_WIDTH, CELL_WIDTH_BORDER } from "constants/calendar";

export function CalendarMonthCellText({ text, length }) {
  return (
    <p
      className="m-0 calendar-month-cell-text"
      style={{ maxWidth: length * (CELL_WIDTH - CELL_WIDTH_BORDER) }}
    >
      {text}
    </p>
  );
}

CalendarMonthCellText.propTypes = {
  text: PropTypes.string,
  length: PropTypes.number,
};

CalendarMonthCellText.defaultProps = {
  text: "",
  length: 0,
};
