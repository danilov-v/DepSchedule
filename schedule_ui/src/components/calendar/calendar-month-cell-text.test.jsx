import React from "react";
import { shallow } from "enzyme";
import { CalendarMonthCellText } from "./calendar-month-cell-text";

describe("calendar-month-cell-text", () => {
  describe("<CalendarMonthCellText />", () => {
    test("should render and match snapshot", () => {
      const component = shallow(
        <CalendarMonthCellText text="test" length={5} />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
