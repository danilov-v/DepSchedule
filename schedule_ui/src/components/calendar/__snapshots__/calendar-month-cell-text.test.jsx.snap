// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calendar-month-cell-text <CalendarMonthCellText /> should render and match snapshot 1`] = `
<p
  className="m-0 calendar-month-cell-text"
  style={
    Object {
      "maxWidth": 170,
    }
  }
>
  test
</p>
`;
